src/DIRAC/ConfigurationSystem/Client/Helpers/Resources.py

```diff
@@ -470,3 +472,172 @@ def getInfoAboutProviders(of=None, providerName=None, option='', section=''):
+        if not result['OK']:
+          result = gConfig.getSections('/Resources/Sites/%s/%s/Cloud/%s/Images' % (grid, site, ce))
+          if not result['OK']:
+            return result
+        vmTypes = result['Value']
+        for vmType in vmTypes:
+          if vmTypeList is not None and vmType not in vmTypeList:
+            continue
+          if vo:
+            voList = gConfig.getValue('/Resources/Sites/%s/%s/Cloud/%s/VMTypes/%s/VO' % (grid, site, ce, vmType), [])
+            if not voList:
+              voList = gConfig.getValue('/Resources/Sites/%s/%s/Cloud/%s/Images/%s/VO' % (grid, site, ce, vmType), [])
+            if voList and vo not in voList:
+              continue
+          resultDict.setdefault(site, {})
+          resultDict[site].setdefault(ce, ceOptionsDict)
+          resultDict[site][ce].setdefault('VMTypes', {})
+          result = gConfig.getOptionsDict('/Resources/Sites/%s/%s/Cloud/%s/VMTypes/%s' % (grid, site, ce, vmType))
+          if not result['OK']:
+            result = gConfig.getOptionsDict('/Resources/Sites/%s/%s/Cloud/%s/Images/%s' % (grid, site, ce, vmType))
+            if not result['OK']:
+              continue
+          vmTypeOptionsDict = result['Value']
+          resultDict[site][ce]['VMTypes'][vmType] = vmTypeOptionsDict
+
+  return S_OK(resultDict)
+
+
+def getVMTypeConfig(site, ce='', vmtype=''):
+  """ Get the VM image type parameters of the specified queue
+  """
+  tags = []
+  grid = site.split('.')[0]
+  if not ce:
+    result = gConfig.getSections('/Resources/Sites/%s/%s/Cloud' % (grid, site))
+    if not result['OK']:
+      return result
+    ceList = result['Value']
+    if len(ceList) == 1:
+      ce = ceList[0]
+    else:
+      return S_ERROR('No cloud endpoint specified')
+
+  result = gConfig.getOptionsDict('/Resources/Sites/%s/%s/Cloud/%s' % (grid, site, ce))
+  if not result['OK']:
+    return result
+  resultDict = result['Value']
+  ceTags = resultDict.get('Tag')
+  if ceTags:
+    tags = fromChar(ceTags)
+  resultDict['CEName'] = ce
+
+  if vmtype:
+    result = gConfig.getOptionsDict('/Resources/Sites/%s/%s/Cloud/%s/VMTypes/%s' % (grid, site, ce, vmtype))
+    if not result['OK']:
+      return result
+    resultDict.update(result['Value'])
+    queueTags = resultDict.get('Tag')
+    if queueTags:
+      queueTags = fromChar(queueTags)
+      tags = list(set(tags + queueTags))
+
+  if tags:
+    resultDict['Tag'] = tags
+  resultDict['VMType'] = vmtype
+  resultDict['Site'] = site
+  return S_OK(resultDict)
+
+
+def getPilotBootstrapParameters(vo='', runningPod=''):
+  """ Get all of the settings required to bootstrap a cloud instance. """
+  op = Operations.Operations(vo=vo)
+  result = op.getOptionsDict('Cloud')
+  opParameters = {}
+  if result['OK']:
+    opParameters = result['Value']
+  opParameters['VO'] = vo
+  opParameters['ReleaseProject'] = op.getValue('Cloud/ReleaseProject', 'DIRAC')
+  opParameters['ReleaseVersion'] = op.getValue('Cloud/ReleaseVersion')
```

Comment:
How are these different from what can be found in the CS Pilot section?

Reply:
This sets the version of the other parts of DIRAC (VirtualMachineMonitor) installed in the cloud instance, so does have a different use/meaning to the main pilot version.

Reply:
I see. What if you do
```suggestion
  opParameters['ReleaseProject'] = op.getValue('Cloud/ReleaseProject', 'DIRAC')
  opParameters['ReleaseVersion'] = op.getValue('Cloud/ReleaseVersion', op.getValue('Pilot/Version'))
```
?

Reply:
Sure, I've done that.